reconcile/prometheus_rules_tester.py

```diff
@@ -0,0 +1,269 @@
+import re
+import sys
+import yaml
+import json
+import semver
+import logging
+import traceback
+
+import reconcile.utils.gql as gql
+import reconcile.utils.threaded as threaded
+import reconcile.utils.promtool as promtool
+import reconcile.openshift_resources_base as orb
+
+QONTRACT_INTEGRATION = 'prometheus_rules_tester'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+PROMETHEUS_RULES_PATHS_QUERY = '''
+{
+  resources: resources_v1(schema: "/openshift/prometheus-rule-1.yml") {
+    path
+  }
+}
+'''
+
+PROMETHEUS_RULES_TESTS_QUERY = '''
+{
+  tests: resources_v1(schema: "/app-interface/prometheus-rule-test-1.yml") {
+    path
+    content
+  }
+}
+'''
+
+
+def get_prometheus_tests():
+    '''Returns a path indexed dict with the prometheus tests content'''
+    gqlapi = gql.get_api()
+    tests = {}
+    for t in gqlapi.query(PROMETHEUS_RULES_TESTS_QUERY)['tests']:
+        # This can be a jinja template. We cannot load the yaml here
+        tests[t['path']] = t['content']
+
+    return tests
+
+
+# returned structure:
+# rules = {
+#    'path': {
+#        'cluster_name': {
+#            'namespace': {
+#                'rule_spec': spec
+#                'variables: { ... } # openshift resource variables if any
+#    (...)
+def get_prometheus_rules(cluster_name):
+    '''Returns a dict of dicts indexed by path with rule data'''
+    gqlapi = gql.get_api()
+    rules = {}
+    for r in gqlapi.query(PROMETHEUS_RULES_PATHS_QUERY)['resources']:
+        rules[r['path']] = {}
+
+    for n in gqlapi.query(orb.NAMESPACES_QUERY)['namespaces']:
+        cluster = n['name']
+        namespace = n['cluster']['name']
+
+        if cluster_name and cluster != cluster_name:
+            continue
+
+        if not n['managedResourceTypes'] or \
+           'PrometheusRule' not in n['managedResourceTypes']:
+            continue
+
+        openshift_resources = n.get('openshiftResources')
+        if not openshift_resources:
+            logging.warning('No openshiftResources defined for namespace'
+                            f'{namespace} in cluster {cluster}')
+            continue
+
+        for r in openshift_resources:
+            path = r['path']
+            if path not in rules:
+                continue
+
+            # Or we will get an unexepected and confusing html_url annotation
+            if 'add_path_to_prom_rules' not in r:
```

Comment:
is `add_path_to_prom_rules` ever in `r`?

Reply:
No, but it is set to `True` in `fetch_provider_resource`. That's why I needed to add this and make sure it is taken into account in `orb. fetch_openshift_resource`